src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -603,6 +619,35 @@ internal override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDec
                 {
                     arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerMemberNameAttribute = true;
                 }
+                else if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.CallerArgumentExpressionAttribute))
+                {
+                    // PROTOTYPE(caller-expr): From 333fred review:
+                    // As the language exists today we can do this as an early attribute, but if we do implement the
+                    // nameof proposal (allowing nameof(parameter) in parameters/attributes on methods) I don't believe it
+                    // will end well (this will cause a loop or fail to bind entirely).
+                    // Consider moving binding of the actual attribute data into regular attribute binding.
```

Comment:
>Consider moving binding of the actual attribute data into regular attribute binding. [](start = 23, length = 84)

I think there is a reason why attributes related to default values are bound early, so we cannot move binding of this attribute to a later time, or we should be able to do this for all similar attributes.

Reply:
@AlekseyTs Probably wait until the nameof proposal is implemented, add a test, and see how things goes?

Reply:
> Probably wait until the nameof proposal is implemented, add a test, and see how things goes?

I suggest removing this comment altogether and track it in an issue including the one tracking the other feature (test plan issue?). It looks like we already have other early bound attributes that are going to be affected. 

---
In reply to: [600230155](https://github.com/dotnet/roslyn/pull/51952#discussion_r600230155) [](ancestors = 600230155)

Reply:
@AlekseyTs I removed the comment. Was an issue tracking the nameof implementation opened? I can't find it.

Reply:
@jcouv Could you record this aspect that possibly should be looked at when we allow nameof for parameters in attributes on methods.

> I removed the comment. Was an issue tracking the nameof implementation opened? I can't find it.



---
In reply to: [600717163](https://github.com/dotnet/roslyn/pull/51952#discussion_r600717163) [](ancestors = 600717163)